editoast/src/views/timetable/similar_trains.rs

```diff
@@ -360,21 +370,22 @@ async fn similar_trains(
     // Step 6: determine which similar train to choose for each segment
     // ----------------------------------------------------------------
 
-    let similar_trains =
-        decide_best_train_combination(trains.iter().map(|(_, trains)| trains).cloned().collect());
+    let similar_trains = decide_best_train_combination(
+        trains
+            .iter()
+            .map(|(_, trains)| trains)
+            .filter(|trains| !trains.is_empty())
+            .cloned()
```

Comment:
```suggestion
```

nit: I believe we can get rid of this if we change the parameter to `Vec<&HashSet<...>>`